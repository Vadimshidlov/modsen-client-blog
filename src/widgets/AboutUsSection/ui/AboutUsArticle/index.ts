export { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle';
