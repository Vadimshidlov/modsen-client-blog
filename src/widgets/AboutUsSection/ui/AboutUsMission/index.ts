export { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission/AboutUsMission';
