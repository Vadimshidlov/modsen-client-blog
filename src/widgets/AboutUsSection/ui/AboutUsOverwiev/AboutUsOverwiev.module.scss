.owerviewContainer {
  position: relative;

  .owerviewImage {
    width: 100%;
    object-fit: cover;
  }

  .owerviewDataContainer {
    position: absolute;
    display: flex;
    column-gap: 68px;
    bottom: 23px;
    left: 9%;
    background: $yellow;
    padding: 32px 48px;

    @media (max-width: $tablet) {
      padding: 16px 24px;
      column-gap: 0px;
      justify-content: space-between;
    }

    @media (max-width: 600px) {
      flex-direction: column;
      align-items: center;
      row-gap: 10px;
    }

    .owerviewDataItem {
      display: flex;
      flex-direction: column;

      .owerviewDataTitle {
        @include display;

        @media (max-width: $tablet) {
          @include headingThree;
        }
      }
    }

    @media (max-width: 1140px) {
      position: static;
    }
  }

  .colorBarContainer {
    position: absolute;
    left: 9%;
    bottom: 0;
    height: 23px;
    width: 68%;
    display: flex;
    flex-direction: row-reverse;
    justify-content: flex-end;

    @media (max-width: 1140px) {
      position: static;
      width: 100%;
    }

    .firstColorBar {
      height: 23px;
      max-width: 780px;
      width: 100%;
      background: $yellow;
    }

    .secondColorBar {
      height: 23px;
      max-width: 282px;
      width: 100%;
      background: $lightFiolet;
    }
  }
}
