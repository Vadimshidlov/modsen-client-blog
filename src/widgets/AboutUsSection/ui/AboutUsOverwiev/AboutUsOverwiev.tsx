import React from 'react';
import styles from '@/widgets/AboutUsSection/ui/AboutUsOverwiev/AboutUsOverwiev.module.scss';
import Image from 'next/image';
import owerviewImageSrc from '@/widgets/AboutUsSection/assets/aboutUsOwerview.png';
import { useTranslations } from 'next-intl';

export function AboutUsOverwiev() {
  const t = useTranslations('AboutUsOverwiev');

  return (
    <section className={styles.owerviewContainer}>
      <Image className={styles.owerviewImage} src={owerviewImageSrc} alt="owerview-image" />
      <div className={styles.owerviewDataContainer}>
        <div className={styles.owerviewDataItem}>
          <span className={styles.owerviewDataTitle}>12+</span>
          {t('blogsMessage')}
        </div>
        <div className={styles.owerviewDataItem}>
          <span className={styles.owerviewDataTitle}>18K+</span>
          {t('viewMessage')}
        </div>
        <div className={styles.owerviewDataItem}>
          <span className={styles.owerviewDataTitle}>30K+</span>
          {t('userMessage')}
        </div>
      </div>
      <div className={styles.colorBarContainer}>
        <div className={styles.firstColorBar} />
        <div className={styles.secondColorBar} />
      </div>
    </section>
  );
}
