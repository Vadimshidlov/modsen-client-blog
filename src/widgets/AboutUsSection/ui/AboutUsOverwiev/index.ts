export { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev/AboutUsOverwiev';
