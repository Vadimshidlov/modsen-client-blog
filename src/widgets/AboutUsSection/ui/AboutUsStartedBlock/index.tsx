export { AboutUsStartedBlock } from '@/widgets/AboutUsSection/ui/AboutUsStartedBlock/AboutUsStartedBlock';
