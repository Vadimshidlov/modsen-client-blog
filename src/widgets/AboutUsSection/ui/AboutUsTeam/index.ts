export { AboutUsTeam } from '@/widgets/AboutUsSection/ui/AboutUsTeam/AboutUsTeam';
