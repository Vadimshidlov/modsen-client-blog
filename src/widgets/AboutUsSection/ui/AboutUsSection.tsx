import React from 'react';
import { AboutUsTitle } from '@/widgets/AboutUsSection/ui/AboutUsTitle';
import { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev';
import { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission';
import { AboutUsTeam } from '@/widgets/AboutUsSection/ui/AboutUsTeam';
import { AboutUsStartedBlock } from '@/widgets/AboutUsSection/ui/AboutUsStartedBlock';
import styles from '@/widgets/AboutUsSection/ui/AboutUsSection.module.scss';
import container from '@/shared/styles/container.module.scss';

export function AboutUsSection() {
  return (
    <section className={`${styles.aboutUsSection} ${container.container}`}>
      <AboutUsTitle />
      <AboutUsOverwiev />
      <AboutUsMission />
      <AboutUsTeam />
      <AboutUsStartedBlock />
    </section>
  );
}
