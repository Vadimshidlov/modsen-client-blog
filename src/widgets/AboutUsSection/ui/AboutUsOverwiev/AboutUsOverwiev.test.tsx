import { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev';
import { render } from '@testing-library/react';

jest.mock('next/font/google', () => ({
  Inter: () => ({ variable: 'fake-variable' }),
}));

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('AboutUsOverwiev tests', () => {
  test('should render overview section with correct data', () => {
    const { getByAltText, getByText } = render(<AboutUsOverwiev />);

    expect(getByAltText('owerview-image')).toBeInTheDocument();

    expect(getByText('12+')).toBeInTheDocument();
    expect(getByText('Mocked blogsMessage')).toBeInTheDocument();

    expect(getByText('18K+')).toBeInTheDocument();
    expect(getByText('Mocked viewMessage')).toBeInTheDocument();

    expect(getByText('30K+')).toBeInTheDocument();
    expect(getByText('Mocked userMessage')).toBeInTheDocument();
  });

  test('should matches snapshot', () => {
    const { container } = render(<AboutUsOverwiev />);
    expect(container).toMatchSnapshot();
  });
});
