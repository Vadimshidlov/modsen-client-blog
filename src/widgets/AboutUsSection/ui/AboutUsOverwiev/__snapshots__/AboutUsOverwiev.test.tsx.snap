// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUsOverwiev tests should matches snapshot 1`] = `
<div>
  <section
    class="owerviewContainer"
  >
    <img
      alt="owerview-image"
      class="owerviewImage"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <div
      class="owerviewDataContainer"
    >
      <div
        class="owerviewDataItem"
      >
        <span
          class="owerviewDataTitle"
        >
          12+
        </span>
        Mocked blogsMessage
      </div>
      <div
        class="owerviewDataItem"
      >
        <span
          class="owerviewDataTitle"
        >
          18K+
        </span>
        Mocked viewMessage
      </div>
      <div
        class="owerviewDataItem"
      >
        <span
          class="owerviewDataTitle"
        >
          30K+
        </span>
        Mocked userMessage
      </div>
    </div>
    <div
      class="colorBarContainer"
    >
      <div
        class="firstColorBar"
      />
      <div
        class="secondColorBar"
      />
    </div>
  </section>
</div>
`;
